### DISCUSSION TITLE
Dark theme for not hurting our eyes at movie nights

### DISCUSSION THREAD

**Discussion by sxiii** [Syncplay/syncplay#480] [Archive]
This topic would be extremely important for me. When I switch from the black movie background to chat sometimes or configuration; white background kicking eyes off. Would be nice to have a globally "dark mode".

**Comment by Et0h** [comment]
At present Syncplay mostly uses the PySide default colours which basically uses the style that Windows tells it to use, so for example if you enable a high contrast display with white-on-black then that is what Syncplay will use.

If someone wanted to add a --dark-mode command line option and to set it to use a different colour scheme then I would be happy for them to make the PR for this as I agree that it would be a nice feature.

**Comment by sxiii** [comment]
@Et0h never used SyncPlay on Windows, only on Manjaro and Ubuntu. :dagger: 

--dark-mode would be really nice.

**Comment by Et0h** [comment]
@sxiii Ah, you're right. I use Syncplay on Windows so that's what I thought of first. By Windows I should have said 'the OS' (or is it 'the desktop environment'?).

**Comment by sxiii** [comment]
@Et0h right! =)) Maybe I would try adding a dark mode set if you can lead me through places that are used in PySide / the toolkit used for develop the app? Thanks

**Comment by CalMaciborka** [comment]
Just gonna say a dark mode on Windows would be very nice. Very jealous of the MacOS dark mode.

**Comment by sxiii** [comment]
Maybe we should start picking up tips so we can give some BATs or other crypto to actual developer who will make a --dark-mode for Syncplay. :)

**Comment by daniel-123** [comment]
@sxiii: I took a look at it since Syncplay never used anything specific in terms of theming. Because it's using Qt, it also uses whatever theme Qt has set for it. This means on Qt based desktop environments it will use the system theme.

If you aren't using Qt based desktop environment you can try using [qt5ct](https://sourceforge.net/projects/qt5ct/) which I [just confirmed that works](https://i.imgur.com/osKjC2S.png). Or any of the other methods outlined [here on Arch Wiki](https://wiki.archlinux.org/index.php/Uniform_look_for_Qt_and_GTK_applications).

It probably wouldn't be particularly hard to code Syncplay specific theme dialog, but I'm not entirely sure if it's worthwhile given that standard tools already provide that functionality.

**Reply by 0lhi** [reply]
Is there a way to setup a Dark Theme for Syncplay without affecting other QT5 applications?

`Exec=env QT_STYLE_OVERRIDE=cde syncplay`

Entering the above in a Terminal gives it a different QT5 Style, but the Color Palette remains a bright one.

**Comment by sxiii** [comment]
Oh great @daniel-123 . Thank you very much for pointing out. I'll check the qt5ct for sure! Maybe at least it worth mentioning on how to change the theme for both cases (using Qt and not using Qt env).

EDIT: I realized that I had qt5ct installed in my Manjaro. And you're right it works -> but only for the SyncPlay installed into the system (say; AUR repos). When I've tried it with AppImage it didn't work; which seems to say that AppImage uses included somewhere else settings or something like that. Not a really big deal, of course.

**Comment by CalMaciborka** [comment]
I'd actually pay good money for a dark mode on Windows. There are definitely many people I know that would love it.

**Comment by sxiii** [comment]
> I'd actually pay good money for a dark mode on Windows. There are definitely many people I know that would love it.

@CalMaciborka

You can do two things: first, try to compile qt5ct on windows and see if it works (I know it should be possible to integrate due to KDE having some windows versions) or maybe there is another qt theme control panel for windows out there. Second, you might just ditch windows or install any lightweight linux in parallel just for the watching movies together or even cook a live USB flash for this goal and have a dark theme with this qt5ct. It should be even pretty easy to even write own theme now. Or just use the default ones/find ready on the theme-ing websites. :)

It looks pretty cool even with default themes:
![image](https://user-images.githubusercontent.com/575925/78315542-9cb90180-755d-11ea-9197-57938e984e78.png)

**Comment by sxiii** [comment]
@CalMaciborka have you managed to run dark theme in windows finally? :)

**Comment by Zefferis** [comment]
Just gonna give a quick thumbs up to the idea of a dark mode for windows 👀

**Comment by Et0h** [comment]
Windows users: Please test out PR #483 and let me know if it works for you. Light/dark mode should follow your system settings for Windows 10 Anniversary users and above.

**Reply by Zefferis** [reply]
Looks and behaves great! 

Is there a particular limitation that prevents the top title bar of Syncplay also obtaining a dark mode colorset? 

I saw something similar come up during a discussion of MPV Darkmode

It referenced:
https://github.com/ysc3839/win32-darkmode 
(Example application shows how to use undocumented dark mode API introduced in Windows 10 1809.)

under:
https://github.com/mpv-player/mpv/issues/6901 (Windows Dark Theme Support) 
Examples were provided showing 32bit programs with dark title bars
______
![image](https://user-images.githubusercontent.com/15270437/141655751-51b082b3-48db-4e83-8433-5b8680abbb5d.png)

**Comment by Et0h** [comment]
@Zefferis I'm not sure if I like the idea of using an undocumented API. The way I make the title bar darker is to set a dark grey accent under the 'Colours' settings and then set 'Show the accent colour on the following surfaces' -> 'Title bars and window borders' to true. I think that only works if the window is highlighted, though.

**Reply by Zefferis** [reply]
Sounds good, I'll test whatever y'all got 👍

**Comment by Et0h** [comment]
If there was a well-maintained Python module/library/package released under an appropriate license which Syncplay could use which made use of the undocumented Windows API to enable dark mode then that might be more palatable than just using an undocumented API which nobody is trying to support.
